api/include/opentelemetry/std/span.h

```diff
@@ -26,13 +26,16 @@
 #      define HAVE_SPAN
 #    endif
 #  endif
-#  if __has_include(<span>) && !defined(HAVE_SPAN)  // Check for span
-#    define HAVE_SPAN
-#  endif
-#  if !__has_include(<string_view>)  // Check for string_view
-#    error \
-        "STL library does not support std::span. Possible solution:"                   \
-         " - #undef HAVE_CPP_STDLIB // to use OpenTelemetry nostd::string_view"
+#  if !defined(HAVE_SPAN)
+#    // Check for Visual Studio span
+#    if defined(_MSVC_VALUE) && _HAS_CXX20
```

Comment:
Where is `_MSVC_VALUE` defined?

Reply:
`_MSVC_VALUE` is a predefined macro added since Visual Studio 2017+ and is [described here](https://docs.microsoft.com/en-us/cpp/preprocessor/predefined-macros?view=msvc-160)

`_MSVC_LANG Defined as an integer literal that specifies the C++ language standard targeted by the compiler. It's set only in code compiled as C++. The macro is the integer literal value 201402L by default, or when the /std:c++14 compiler option is specified. The macro is set to 201703L if the /std:c++17 compiler option is specified. It's set to a higher, unspecified value when the /std:c++latest option is specified. Otherwise, the macro is undefined. The _MSVC_LANG macro and /std (Specify Language Standard Version) compiler options are available beginning in Visual Studio 2015 Update 3.` ... 

Based on my understanding, we can check that macro is defined. Means we may get native `std::span` support. And if it is defined (means we are running with at least Visual Studio 2017+), we subsequently check if `_HAS_CXX20=1`, if the build tools have support for `c++20`, and only then we may `#include <span>`. If we don't do that check and simply include the `<span>` if it exists, then msvc compiler fails with an error.

Reply:
Good catch! It was one of those moments. Fixed it.